lisa/features/serial_console.py

```diff
@@ -33,38 +33,64 @@ class PanicInfo:
-        # Synchronous Exception at 0x000000003FD04000
-        re.compile(r"^(.*Synchronous Exception at.*)$", re.MULTILINE),
-        # Lockup patterns
-        re.compile(r"^(.*soft lockup.*)$", re.MULTILINE | re.IGNORECASE),
-        re.compile(r"^(.*hard lockup.*)$", re.MULTILINE | re.IGNORECASE),
-        # BUG and kernel errors
+        re.compile(r"^(.*panic:.*)$", re.MULTILINE),
+        # Oops (kernel crash)
+        re.compile(r"^(.*Oops:.*)$", re.MULTILINE),
+        # Critical RCU stalls (both keywords must appear for strong signal)
+        re.compile(
+            r"^(.*rcu_sched self-detected stall.*)$", re.MULTILINE | re.IGNORECASE
+        ),
+        # Soft lockup with CPU number (strong marker)
+        re.compile(
+            r"^(.*soft lockup - CPU#\d+ stuck.*)$", re.MULTILINE | re.IGNORECASE
+        ),
+        # Hard lockup with CPU number (strong marker)
+        re.compile(
+            r"^(.*hard lockup - CPU#\d+ stuck.*)$", re.MULTILINE | re.IGNORECASE
+        ),
+        # BUG with specific location (stronger than generic BUG)
         re.compile(r"^(.*BUG:.*)$", re.MULTILINE),
+        # Kernel NULL pointer dereference
         re.compile(r"^(.*kernel NULL pointer.*)$", re.MULTILINE | re.IGNORECASE),
+        # Unable to handle kernel paging request
         re.compile(r"^(.*unable to handle.*)$", re.MULTILINE | re.IGNORECASE),
-        # Hung tasks and watchdog
-        re.compile(r"^(.*hung task.*)$", re.MULTILINE | re.IGNORECASE),
-        re.compile(r"^(.*watchdog.*)$", re.MULTILINE | re.IGNORECASE),
-        # RCU stalls
-        re.compile(r"^(.*rcu_sched.*)$", re.MULTILINE | re.IGNORECASE),
-        re.compile(r"^(.*stall.*)$", re.MULTILINE | re.IGNORECASE),
+        # GRUB prompt (system failed to boot kernel)
+        re.compile(r"^(.*grub>.*)$", re.MULTILINE),
+        # OS halted
+        re.compile(r"^The operating system has halted.$", re.MULTILINE),
+        # Synchronous Exception (firmware/UEFI crashes)
+        re.compile(r"^(.*Synchronous Exception at.*)$", re.MULTILINE),
     ]
 
-    # ignore some return lines, which shouldn't be a panic line.
+    # Ignore list - harmless patterns that contain panic-like keywords
     panic_ignorable_patterns: List[Pattern[str]] = [
+        # ClusterIP module load message
         re.compile(
             r"^(.*ipt_CLUSTERIP: ClusterIP.*loaded successfully.*)$", re.MULTILINE
         ),
-        # This is a known issue with Hyper-V when running on AMD processors.
-        # The problem occurs in VM sizes that have 16 or more vCPUs which means 2 or
-        # more NUMA nodes on AMD processors.
-        # The call trace is annoying but does not affect correct operation of the VM.
+        # Known Hyper-V topology warning on AMD (harmless call trace)
         re.compile(r"(.*RIP: 0010:topology_sane.isra.*)$", re.MULTILINE),
+        # NMI watchdog disabled message (informational, not a panic)
+        re.compile(
```

Comment:
It's helpful to explain why something is being ignored so others can understand the reasoning. Including a quick example makes it even clearer.